src/LanguageServer/ExternalAccess/Copilot/Handler/CopilotRequestContext.cs

```diff
@@ -0,0 +1,21 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.CodeAnalysis.LanguageServer.Handler;
+
+namespace Microsoft.CodeAnalysis.LanguageServer.ExternalAccess.Copilot;
+
+/// <summary>
+/// Context for requests handled by handlers like <see cref="AbstractCopilotLspServiceDocumentRequestHandler{TRequest, TResponse}"/>
+/// </summary>
+internal readonly struct CopilotRequestContext(RequestContext context)
+{
+    /// <inheritdoc cref="RequestContext.Solution"/>
+    public Solution? Solution => context.Solution;
```

Comment:
since we have requireslspsolution this should always be not null (would potentially throw here if context.Solution is null)

Reply:
```suggestion
    public Solution Solution => context.Solution ?? throw new InvalidOperationException();
```
You meant something like this?

Reply:
yup